spark/src/main/java/org/apache/iceberg/actions/BaseSparkAction.java

```diff
@@ -86,6 +126,32 @@
     return manifestDF.union(otherMetadataFileDF).union(manifestListDF);
   }
 
+  protected static Dataset<Row> loadMetadataTable(SparkSession spark, String tableName, String tableLocation,
+                                                  MetadataTableType type) {
+    DataFrameReader noCatalogReader = spark.read().format("iceberg");
+    if (tableName.contains("/")) {
+      // Metadata location passed, load without a catalog
+      return noCatalogReader.load(tableName + "#" + type);
+    } else {
+      // Try catalog based name based resolution
+      try {
+        return spark.table(tableName + "." + type);
+      } catch (Exception e) {
```

Comment:
Can this be a more specific exception? Seems dangerous to catch 'em all.

Reply:
:pokemon: - Let me check what gets thrown and I"ll try to narrow it down to reasonable classes of exception

Reply:
Unfortunately these are unchecked from Scala so we do need to catch all runtime exceptions, but I can do a check after the exception is caught if it is a ParseException (2.4) or AnalysisException(3.0) and rethrow if it isn't.